xtask/src/openapi.rs

```diff
@@ -0,0 +1,63 @@
+use std::path::Path;
+use std::process::Command;
+use std::{env, fs};
+
+use anyhow::anyhow;
+use clap::Parser;
+
+use trustify_server::openapi::openapi;
+
+#[derive(Debug, Parser)]
+pub struct Validate {}
+
+impl Validate {
+    pub fn run(self) -> anyhow::Result<()> {
+        let command = if command_exists("podman") {
+            "podman"
+        } else if command_exists("docker") {
+            "docker"
+        } else {
+            return Err(anyhow!("Neither podman nor docker is installed"));
+        };
+
+        let out_dir = env::temp_dir();
+        let openapi_yaml = Path::new(&out_dir).join("openapi.yaml");
+
+        println!("Generating openapi.yaml at {:?}", openapi_yaml);
+
+        let doc = openapi()
+            .to_yaml()
+            .map_err(|_| anyhow!("Failed to convert openapi spec to yaml"))?;
+
+        fs::write(openapi_yaml, doc).map_err(|_| anyhow!("Failed to write openapi spec"))?;
+
+        // run the openapi generator validator container
+        if Command::new(command)
+            .args([
+                "run",
+                "--rm",
+                "-v",
+                ".:/src",
+                "docker.io/openapitools/openapi-generator-cli:v7.7.0",
+                "validate",
+                "-i",
+                "/src/openapi.yaml",
+            ])
+            .current_dir(out_dir.to_str().unwrap())
+            .status()
+            .map_err(|_| anyhow!("Failed to validate openapi.yaml"))?
+            .success()
+        {
+            Ok(())
+        } else {
+            Err(anyhow!("Failed to validate openapi.yaml"))
```

Comment:
I could not tell if this is going to emit specific validation errors to be able to diagnose the problem.

Reply:
Actual validation errors are logged to console by the container.